scene/gui/code_edit.cpp

```diff
@@ -591,27 +593,40 @@ void CodeEdit::_handle_unicode_input_internal(const uint32_t p_unicode) {
 	if (auto_brace_completion_enabled) {
 		int cl = get_caret_line();
 		int cc = get_caret_column();
-		int caret_move_offset = 1;
 
-		int post_brace_pair = cc < get_line(cl).length() ? _get_auto_brace_pair_close_at_pos(cl, cc) : -1;
+		bool try_to_wrap_selection_with_braces = (had_selection && auto_brace_completion_enabled);
 
-		if (has_string_delimiter(chr) && cc > 0 && _is_char(get_line(cl)[cc - 1]) && post_brace_pair == -1) {
-			insert_text_at_caret(chr);
-		} else if (cc < get_line(cl).length() && _is_char(get_line(cl)[cc])) {
-			insert_text_at_caret(chr);
-		} else if (post_brace_pair != -1 && auto_brace_completion_pairs[post_brace_pair].close_key[0] == chr[0]) {
-			caret_move_offset = auto_brace_completion_pairs[post_brace_pair].close_key.length();
-		} else if (is_in_comment(cl, cc) != -1 || (is_in_string(cl, cc) != -1 && has_string_delimiter(chr))) {
+		if (try_to_wrap_selection_with_braces) {
```

Comment:
`auto_brace_completion_enabled` will always be true here, can simplify to `if (had_selection) {`

Reply:
Updated